src/ralph/backends/data/async_ws.py

```diff
@@ -0,0 +1,176 @@
+"""Websocket stream backend for Ralph."""
+
+from typing import AsyncIterator, Optional, Union
+
+import websockets
+from pydantic import AnyUrl, PositiveInt
+
+from ralph.backends.data.base import (
+    BaseAsyncDataBackend,
+    BaseDataBackendSettings,
+    BaseQuery,
+    DataBackendStatus,
+)
+from ralph.conf import BaseSettingsConfig
+from ralph.exceptions import BackendException
+from ralph.utils import async_parse_iterable_to_dict
+
+
+class WSDataBackendSettings(BaseDataBackendSettings):
+    """Websocket data backend default configuration.
+
+    Attributes:
+        URI (str): The URI to connect to.
+    """
+
+    class Config(BaseSettingsConfig):
+        """Pydantic Configuration."""
+
+        env_prefix = "RALPH_BACKENDS__DATA__WS__"
+
+    URI: AnyUrl
+
+
+class AsyncWSDataBackend(BaseAsyncDataBackend[WSDataBackendSettings, BaseQuery]):
+    """Websocket stream backend."""
+
+    name = "async_ws"
+
+    def __init__(self, settings: Optional[WSDataBackendSettings] = None):
+        """Instantiate the async websocket data backend.
+
+        Args:
+            settings (WSDataBackendSettings or None): The websocket backend settings.
+                If `settings` is `None`, a default settings instance is used instead.
+        """
+        super().__init__(settings)
+        self._client = None
+
+    async def client(self) -> websockets.WebSocketClientProtocol:
+        """Create a websocket client connected to `settings.URI` if it doesn't exist."""
+        if not self._client:
+            try:
+                self._client = await websockets.connect(self.settings.URI)
+            except (websockets.WebSocketException, OSError, TimeoutError) as error:
+                msg = "Failed open websocket connection for %s: %s"
+                self.logger.error(msg, self.settings.URI, error)
+                raise BackendException(msg % (self.settings.URI, error)) from error
+        return self._client
+
+    async def status(self) -> DataBackendStatus:
+        """Implement data backend checks (e.g. connection, cluster status).
+
+        Return:
+            DataBackendStatus: The status of the data backend.
+        """
+        try:
+            await self.client()
+        except BackendException:
```

Comment:
Do you think it will work if the backend has already been instantiated?

Reply:
Why not trying a [`ping`](https://websockets.readthedocs.io/en/stable/reference/asyncio/client.html#websockets.client.WebSocketClientProtocol.ping) instead?

Reply:
Nice catch) Thanks) Indeed, if the connection breaks after client instantiation, this does not work anymore) Updated using ping)